lib/eslint.js

```diff
@@ -617,33 +657,45 @@ module.exports = (function() {
             Object.keys(config.rules).filter(function(key) {
                 return getRuleSeverity(config.rules[key]) > 0;
             }).forEach(function(key) {
-
-                var ruleCreator = rules.get(key),
-                    severity = getRuleSeverity(config.rules[key]),
-                    options = getRuleOptions(config.rules[key]),
-                    rule;
-
-                if (ruleCreator) {
-                    try {
-                        rule = ruleCreator(new RuleContext(
-                            key, api, severity, options,
-                            config.settings, config.ecmaFeatures
-                        ));
-
-                        // add all the node types as listeners
-                        Object.keys(rule).forEach(function(nodeType) {
-                            api.on(nodeType, timing.enabled
-                                ? timing.time(key, rule[nodeType])
-                                : rule[nodeType]
-                            );
-                        });
-                    } catch(ex) {
-                        ex.message = "Error while loading rule '" + key + "': " + ex.message;
-                        throw ex;
+                var ruleCreator,
+                    severity,
+                    options,
+                    rule,
+                    deprecationMsg;
+
+                if (!rules.get(key)) {
```

Comment:
When I consider deprecation, there would be a scenario where the rule is still present but will be deprecated in the future. Thats what deprecation mean otherwise the rule would purely be unknown.
Are we not tacking that scenario?